Bottom: 1704bb412fc06d970f1387e4926764133395b57b
Top:    3e3724434caedf6f286d2f8dd54bdc4526664733
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-20 00:51:16 +0000

Refresh of play-with-glsl-shader-for-rend

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index abedaad..0a45934 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -937,34 +937,7 @@ load_built_in_shaders (void)
           "  gl_FragColor = gl_Color;\n"
           "}\n";
 
-  char *resistor_fs_source =
-          "uniform sampler1D detail_tex;\n"
-          "uniform sampler2D bump_tex;\n"
-          "\n"
-          "void main()\n"
-          "{\n"
-          "  vec3 bumpNormal = texture2D (bump_tex, gl_TexCoord[1].st).rgb;\n"
-          "  vec3 detailColor = texture1D (detail_tex, gl_TexCoord[0].s).rgb;\n"
-          "\n"
-          "  /* Uncompress vectors ([0, 1] -> [-1, 1]) */\n"
-          "  vec3 lightVectorFinal = -1.0 + 2.0 * gl_Color.rgb;\n"
-          "  vec3 halfVectorFinal = -1.0 + 2.0 * gl_TexCoord[2].xyz;\n"
-          "  vec3 bumpNormalVectorFinal = -1.0 + 2.0 * bumpNormal;\n"
-          "\n"
-          "  /* Compute diffuse factor */\n"
-          "  float diffuse = clamp(dot(bumpNormalVectorFinal,\n"
-          "                            lightVectorFinal),0.0, 1.0);\n"
-          "  float specular = pow(clamp(dot(bumpNormalVectorFinal,\n"
-          "                                 halfVectorFinal), 0.0, 1.0),\n"
-          "                       2.0);\n"
-          "  specular *= 0.4;\n"
-          "\n"
-          "   gl_FragColor = vec4(detailColor * (0.3 + 0.7 * diffuse) + \n"
-          "                    vec3(specular, specular, specular), 1.0);\n"
-          "}\n";
-
   circular_program = hidgl_shader_new ("circular_rendering", NULL, circular_fs_source);
-  resistor_program = hidgl_shader_new ("resistor_rendering", NULL, resistor_fs_source);
 
   hidgl_shader_activate (circular_program);
 }
